cpp/src/copying/slice.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 BlazingDB, Inc.
+ *     Copyright 2019 Christian Noboa Mardini <christian@blazingdb.com>
+ *     Copyright 2019 William Scott Malpica <william@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/cudf.h>
+#include <utilities/error_utils.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/copying.hpp>
+#include <algorithm>
+
+namespace cudf {
+
+namespace experimental {
+
+namespace detail {
+
+std::vector<cudf::column_view> slice(cudf::column_view const& input,
+                                                std::vector<size_type> const& indices){
+
+    CUDF_EXPECTS(indices.size()%2 == 0, "indices size must be even");
+    std::vector<cudf::column_view> result{};
+
+    if(indices.size() == 0 or input.size() == 0) {
+        return result;
+    }
+
+    std::vector<std::pair<size_type, size_type>> indices_tuple{};
+
+    auto it_start = indices.begin();
+    auto it_end = indices.begin() + 1;
+
+    for(;it_start != indices.end(); std::advance(it_start, 2), std::advance(it_end, 2)) { 
```

Comment:
This is off by one. I think you want this:
```suggestion
    for(;it_end != indices.end(); std::advance(it_start, 2), std::advance(it_end, 2)) { 
```

Reply:
Also, I think this may be a case where a raw for loop is actually better than an algorithm given the tools available to use in C++14:
```c++
for(size_t i = 0; i < indices.size(); i+=2){
   result.emplace_back(slice(input, indices[i], indices[i+1]);
}
```

This could be done with a `transform` using a `strided_iterator` for the input, but that's not currently available in C++14 without implementing ourselves.

Reply:
Yes, I was trying to get an iterator to do that, but couldn't find one.